### PR TITLE
fix: App stuck when open event from favorites

### LINKED ISSUE #631: App stuck when open event from favorites
**Describe the bug**

The app is stuck when open event details from favorites. Back pressed is not working and title changes on back pressed.

**To Reproduce**
Remove the fragment on back pressed.

**Screenshots**

gifmaker_20181011025533

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |Mi Max 2|
|Android Version|Nogute 7.1    |

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #631

Changes: MainActivity.kt

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 1]
```diff
@@ -127,9 +127,14 @@ class MainActivity : AppCompatActivity() {
...
+        val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
         if (currentFragment !is EventsFragment) {
-            loadFragment(EventsFragment())
-            navigation.selectedItemId = navigation_events
+            if (rootFragment is EventDetailsFragment) {
+                super.onBackPressed()
+            } else {
+                loadFragment(EventsFragment())
+                navigation.selectedItemId = navigation_events
+            }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: This is getting too complicated. Think of a different solution

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Updated.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: > This is getting too complicated. Think of a different solution

Updated. please review.

**Comment by liveHarshit** [comment]
@nikit19 No, because this problem is in ticket details fragment which is placed on rootLayout but #629 is for event fragment which placed on frameContainer.

**Comment by liveHarshit** [comment]
@iamareebjamal Please review it.

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 2]
```diff
@@ -127,7 +127,8 @@ class MainActivity : AppCompatActivity() {
 
     override fun onBackPressed() {
         val currentFragment = this.supportFragmentManager.findFragmentById(R.id.frameContainer)
-        if (currentFragment !is EventsFragment) {
+        val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
+        if (currentFragment !is EventsFragment && rootFragment !is EventDetailsFragment) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: This is also special casing but OK, it was already there

**Pull request merged**